### ISSUE TITLE
download_url param doesn't use $version as passed, in, always uses default in params::version

### ISSUE THREAD

**Issue by sjoeboo** [claranet/puppet-consul_template#31]
Just tested and grabbed the resulting catalog: 

```
parameters": {
"purge_config_dir": true,
"bin_dir": "/usr/local/bin",
"arch": "amd64",
"version": "0.10.0",
"install_method": "url",
"os": "linux",
"download_url": "https://github.com/hashicorp/consul-template/releases/download/v0.9.0/consul-template_0.9.0_linux_amd64.tar.gz",
"package_name": "consul-template",
```

I don't think a param can use the values from other params. I can pass in the download url for the version i need, but as is this doesn't work as intended.

**Comment by gdhbashton** [comment]
I've bumped the default module version to 0.10.0 (latest version of upstream, released months ago...) - that'll have to do for now. I did find a similar problem with the `consul` module yesterday and it has changed its install.pp as a result. I need to apply the same pattern here but have no time. PR would always be welcome if you want to have a go?

**Comment by thanandorn** [comment]
@sjoeboo 

This should be fixed in #36

**Comment by gdhbashton** [comment]
True dat.
